### ISSUE TITLE
Invite URL Filtering False Positives

### ISSUE THREAD

**Issue by sco1** [python-discord/bot#230]
Fixes introduced in #226 seem to have exposed an issue with the deblanking used by the invite filter prior to applying the regex:

https://github.com/python-discord/bot/blob/e6d761ecfd1359354ec3889a667c986d40e7376e/bot/cogs/filtering.py#L226-L228

Which produces something like the following:

```python
>>> text = "Can the discord.gg/python invite be mentioned in"
>>> text = text.replace(" ", "")
>>> text = text.replace("\\", "")
>>> invites = re.findall(INVITE_RE, text, re.IGNORECASE)
>>> print(invites)
['pythoninvitebementionedin']
```

So, with the above example we're passing a request to `https://discordapp.com/api/v7/invites/pythoninvitebementionedin`, which returns the following response:

```
{"code": 10006, "message": "Unknown Invite"}
```
Since this JSON doesn't contain a `'guild'` key, it's considered an invite for a Group DM and a mod alert is sent. (Note that before #226 this would have just silently failed)

In theory, this also means that any invite link that is followed by non-blank characters will also throw an alert, even if whitelisted.

**Comment by sco1** [comment]
For a quick look, here are the current API returns:

# Valid Guild
Vanity URLs do not have an `"inviter"` key

```json
{
    "inviter": {
        "username": "Nereus",
        "discriminator": "9479",
        "id": "214483964356853761",
        "avatar": "8d60a53a976baf5905e1f30df52b846b"
    },
    "code": "8NWhsvT",
    "guild": {
        "verification_level": 2,
        "features": [
            "VIP_REGIONS",
            "VANITY_URL",
            "INVITE_SPLASH"
        ],
        "name": "Python",
        "splash": "e7211340f4e830217594b7c0a5e3a349",
        "id": "267624335836053506",
        "icon": "97eb04c7bf4e106ed4c547852e4252ea"
    },
    "channel": {
        "type": 0,
        "id": "267631170882240512",
        "name": "rules-information"
    }
}
```


# Group DM
```json
{
    "code": "aFak3",
    "inviter": {
        "username": "fakeaccount",
        "discriminator": "0000",
        "id": "123456789012345678",
        "avatar": null
    },
    "channel": {
        "icon": null,
        "type": 3,
        "id": "123456789012345678",
        "name": "Test Group DM"
    }
}
```

# Invalid invite
It seems like all invalid invites have a `"code": 10006` key, value pair
```json
{
    "code": 10006,
    "message": "Unknown Invite"
}
```

**Comment by sco1** [comment]
Half of the issue is the straightforward fix of some more robust handling of the API responses, but I'm not sure what to do about the deblanking.

I'm not sure of the context for including it in the first place, is it a significant enough threat to leave in or can we do without it?

**Comment by jos-b** [comment]
@sco1 Looking through the message logs it appears that `d i s c o r d . g g` has never been used apart from when lemon was testing this feature.

**Comment by sco1** [comment]
After looking at this a bit, it seems like the smartest approach for now is to just get rid of the deblanking and addressing any l33t hax0rz should they appear in the future.

Adding more specific handling for the different API responses doesn't really get us anywhere; Discord's API doesn't differentiate between expired and invalid links (and both are equally useless) so there's nothing we can really do with them other than just throw them out, which doesn't need new code.

**Comment by sco1** [comment]
Resolved in #234
